Dockerfile

```diff
@@ -342,87 +342,6 @@ if [[ ${INSTALL_MYSQL_CLIENT:="true"} == "true" ]]; then
 fi
 EOF
 
-# The content below is automatically copied from scripts/docker/install_mssql.sh
```

Comment:
As explained on Slack - we likely do not want to remove this script. This script is to install MSSQL client libraries and pymssql client. Those are not necessarily the same as used by anyone who uses mysql as Airflow Metadata DB.

The pymssql driver installed in this step (together with binary libraries) is used for sure by mssql provider https://github.com/apache/airflow/blob/main/airflow/providers/microsoft/mssql/hooks/mssql.py 

The driver that we recommend to use MSSQL as Metadata DB is a different one (it's `pyodbc` one)

SQL Alchemy might use differnt drivers to communicate with MSSQL https://docs.sqlalchemy.org/en/20/dialects/mssql.html  this is one ofht 

We are testing in CI using `mssql+pyodbc` driver not `mssql+pymssql`. I can't remember the details, but I believe there were issues with using pymssql driver. We might want to take a look at the history of adding MSSQL as core db. See:  https://github.com/apache/airflow/blob/main/scripts/ci/docker-compose/backend-mssql.yml


So the `pymssql` driver is part of the reference image of ours mostly because we want to be able to support `apache-airflow-microsoft-mssql` provider out-of-the-box as we recognise MSSQL as a database our users might often interact with. The goal of this PR is to remove any `MSSQL as Core Airflow Metadatadb backend` - not to remove pymssql library from the image.